### PR TITLE
SAK-43878 better handling when authzgroup does not exist

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

File: sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java [hunk 1]
```diff
@@ -1127,19 +1127,18 @@ public IModel getModel(Object value) {
...
-			for (String s : siteIdWithRoles) {
+		Set<String> roles = new HashSet<String>();
+		for (String s : siteIdWithRoles) {
+			try {
 				Set<Role> roleSet = Locator.getFacade().getAuthzGroupService().getAuthzGroup(s).getRoles();
 				for (Role r : roleSet) {
 					roles.add(r.getId());
 				}
+			} catch (GroupNotDefinedException e) {
+				log.warn("AuthzGroup does not exist, skipping: {}", s);
```
Comment (ern) [inline sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java:1138]: Are we sure you want to see these as warnings in the log? how about debug?

File: sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java [hunk 2]
```diff
@@ -1127,19 +1127,18 @@ public IModel getModel(Object value) {
...
-		List<String> roles = new ArrayList<String>();
-		try {
-			for (String s : siteIdWithRoles) {
+		Set<String> roles = new HashSet<String>();
+		for (String s : siteIdWithRoles) {
+			try {
 				Set<Role> roleSet = Locator.getFacade().getAuthzGroupService().getAuthzGroup(s).getRoles();
 				for (Role r : roleSet) {
 					roles.add(r.getId());
 				}
```
Comment (ern) [inline sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java:1136]: Reduce to one line:
Locator.getFacade().getAuthzGroupService().getAuthzGroup(s).getRoles().forEach(r -> roles.add(r.getId()));

File: sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java (hunk 1 above)
Comment (ottenhoff) [reply sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java:1138]: agree

File: sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java (hunk 2 above)
Comment (ottenhoff) [reply sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java]: done

**Pull request merged**